### PR TITLE
Fix a bug in fdiff(argindex)

### LINKED ISSUE #8510: Differentiation of general functions
```
In [1]:
from sympy import *
var('x, y')
f = Function('f')

In [2]:
f(x + y, x).diff(x)

Out[2]:
Derivative(f(x + y, x), x) + Subs(Derivative(f(_xi_1, x), _xi_1), (_xi_1,), (x + y,))

In [3]:
f(x, x).diff(x)

Out[3]:
2*Derivative(f(x, x), x)
```

For `In [2]`, instead of the result, I would expect, 
`Subs(Derivative(f(x+y, _xi_2), _xi_2), (_xi_2,), (x,)) + Subs(Derivative(f(_xi_1, x), _xi_1), (_xi_1,), (x + y,))`

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #8510

### CODE REVIEW

File: sympy/core/function.py [hunk 1]
```diff
@@ -646,14 +646,25 @@ def fdiff(self, argindex=1):
...
+            use_dummy = False
+            for i in range(len(self.args)):
+                if i == argindex:
+                    continue
+                # See issue 8510
+                if self.args[argindex - 1] in self.args[i].free_symbols:
+                    use_dummy=True
+                    break
+            if not use_dummy:
+                return Derivative(self, self.args[argindex - 1], evaluate=False)
```
Comment (leosartaj) [inline sympy/core/function.py]: You can use ``for else`` construct here. That can save you from declaring ``use_dummy``.

Comment (isuruf) [inline sympy/core/function.py]: Thanks. Didn't know about it before.

**Comment by asmeurer** [comment]
Thanks for the fix.

**Comment by Shekharrajak** [comment]
:+1:

**Pull request merged**